ui/json_formatter.go

```diff
@@ -0,0 +1,220 @@
+package ui
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"time"
+
+	"github.com/gruntwork-io/cloud-nuke/aws"
+	"github.com/gruntwork-io/cloud-nuke/gcp"
+	"github.com/gruntwork-io/cloud-nuke/report"
+	"github.com/gruntwork-io/go-commons/errors"
+)
+
+// GetOutputWriter returns a writer for the specified output file or stdout if empty
+func GetOutputWriter(outputFile string) (io.Writer, func() error, error) {
+	if outputFile == "" {
+		// Return stdout with a no-op closer
+		return os.Stdout, func() error { return nil }, nil
+	}
+
+	file, err := os.Create(outputFile)
+	if err != nil {
+		return nil, nil, errors.WithStackTrace(err)
+	}
+
+	return file, file.Close, nil
+}
+
+// RenderInspectAsJSON renders AWS inspection results as JSON
+func RenderInspectAsJSON(account *aws.AwsAccountResources, query *aws.Query, w io.Writer) error {
+	output := buildInspectOutput(account, query)
+	encoder := json.NewEncoder(w)
+	encoder.SetIndent("", "  ")
+	return encoder.Encode(output)
+}
+
+// RenderGcpInspectAsJSON renders GCP inspection results as JSON
+func RenderGcpInspectAsJSON(project *gcp.GcpProjectResources, w io.Writer) error {
+	output := buildGcpInspectOutput(project)
+	encoder := json.NewEncoder(w)
+	encoder.SetIndent("", "  ")
+	return encoder.Encode(output)
+}
+
+// RenderNukeReportAsJSON renders nuke operation results as JSON
+func RenderNukeReportAsJSON(w io.Writer) error {
+	records := report.GetRecords()
+	generalErrors := report.GetErrors()
+	output := buildNukeOutput(records, generalErrors)
+	encoder := json.NewEncoder(w)
+	encoder.SetIndent("", "  ")
+	return encoder.Encode(output)
+}
+
+// buildInspectOutput constructs the JSON output structure for AWS inspection
+func buildInspectOutput(account *aws.AwsAccountResources, query *aws.Query) InspectOutput {
+	resources := []ResourceInfo{}
```

Comment:
Makes sense to use `var` to initialize a nil list?
`var resources []ResourceInfo`